### PR TITLE
hb-05 task done

### PR DESCRIPTION
**Pull request by less08**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery("from User u where u.email=:email", User.class)
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/dao/UserDaoImpl.java]: you don't need alias here

File: src/main/java/com/dev/cinema/exception/AuthenticationException.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.dev.cinema.exception;
+
+public class AuthenticationException extends Exception {
+    public AuthenticationException(String message) {
+        super(message);
+    }
+
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/exception/AuthenticationException.java]:
```suggestion
```

File: src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isPresent()) {
+            String hashedPassword = HashUtil.hashPassword(password, user.get().getSalt());
+            if (hashedPassword.equals(user.get().getPassword())) {
+                return user.get();
+            }
+        }
+        throw new AuthenticationException("Can't authenticate by email " + email
```
Comment (Max-Pochepets) [inline src/main/java/com/dev/cinema/service/AuthenticationServiceImpl.java]: don't show back the credentials. you might reveal the password while input field hides it

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

**Review by Max-Pochepets — APPROVED** [review]
👍

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**